lib/schema_web/views/page_view.ex

```diff
@@ -122,6 +124,61 @@ defmodule SchemaWeb.PageView do
     Path.basename(name)
   end
 
+  @spec format_class_attribute_source(String.t(), map()) :: String.t()
+  def format_class_attribute_source(class_name, field) do
+    all_classes = Schema.all_classes()
+    source = field[:_source]
+    {ok, path} = build_class_hierarchy(Schema.Utils.to_uid(class_name), source, all_classes)
+
+    if ok do
+      format_class_hierarchy(path, all_classes)
+    else
+      to_string(source)
+    end
+  end
+
+  # Build a class hierarchy path from class to parent_class.
+  # Returns {true, path} when a path is found, and {false, []} otherwise.
+  @spec build_class_hierarchy(atom(), atom(), map(), list()) :: {boolean(), list()}
+  defp build_class_hierarchy(
+         class,
+         parent_class,
+         all_classes,
+         path_result \\ []
+       ) do
+    cond do
+      class == nil ->
+        {false, []}
+
+      class == parent_class ->
+        {true, [class | path_result]}
+
+      true ->
+        build_class_hierarchy(
+          Schema.Utils.to_uid(all_classes[class][:extends]),
+          parent_class,
+          all_classes,
+          [class | path_result]
+        )
+    end
+  end
+
+  defp format_class_hierarchy(path, all_classes) do
+    Enum.map(
+      path,
+      fn class ->
+        class_info = all_classes[class]
+
+        if class_info[:hidden?] do
+          [all_classes[class][:caption], " (hidden class)"]
```

Comment:
I was using the "(hidden class)" bit because the class doesn't show up on the Categories or Classes pages, and I'm thinking it could be confusing for someone looking for the class. I would think a user would think the missing class was a bug.

I'd rather not add "Xxx **Base** Class" though. That's something that could be done better by changing the `caption` of the hidden class.

Reply:
Yeah that's a good point, we can just edit the captions of all base classes to convey the point.